### ISSUE TITLE
Explosion permissions not checking full area

### ISSUE THREAD

**Issue by tethlah** [ForgeEssentials/ForgeEssentialsMain#1587]
I set a spawn area named worldSpawn and using the following command I don't protect it from TNT for some reason:  

/p global zone worldSpawn deny fe.protection.explosion
/p global zone worldSpawn deny fe.protection.break.*
/p global zone worldSpawn deny fe.protection.place.*
/p global zone worldSpawn deny fe.protection.interact.*

None of these seem to stop TNT from being able to be placed at the border and destroy everything in the spawn zone within range.

unrelated note:  For some reason when I set spawn, people can /spawn to it, but when they die they spawn at the default spawn at worldgen.

btw, I'm using build 751.

**Comment by spannerman79** [comment]
A work around for setting spawn is to first obtain your `level.dat` file from your server, use something like [NBTExplorer](http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-tools/1262665-nbtexplorer-nbt-editor-for-windows-and-mac) and then exit the Spawn X, Y Z, save the file and while your sever is shutdown reupload the `level.dat` file
![nbt](https://cloud.githubusercontent.com/assets/7542384/8199230/07702374-14f5-11e5-84ea-359d112f9ea1.PNG)

**Comment by olee** [comment]
The spawn worked some time ago but it really might have stopped working a few versions ago.
I noticed it on my own server as well where I had some problems with players not getting teleported to spawn set by FE.
Will check it these days.

**Comment by tethlah** [comment]
gotcha, so fe doesn't actually protect from explosion damage, it just prevents explosions from occurring within it's borders?

**Comment by olee** [comment]
After looking at the code I wrote, it should actually check the whole explosion area, so it should cancel the explosion if any part of it reaches into the area. However the graphics and sound will always happen on the client, even if the explosion - remember this when testing.

**Comment by tethlah** [comment]
So here is what I did, I set up a 5x5x5 block of glass and protected it using the above settings.  I placed industrial TNT outside of the protected zone but 1 block away from the glass (and thus the protected area).  The resulting explosion destroyed much of the glass inside the protected area, along with the land underneath that was also protected.

**Comment by olee** [comment]
Which is why I said above that it's a bug that it does not work ;-)

**Comment by tethlah** [comment]
oh gotcha, I thought you meant that's what it was doing, not simply what it was supposed to but isn't doing.

**Comment by jogi01** [comment]
we have problems with spawn coords too, new players spawn at roof of spawn building, coords was set in level.dat and in FE _ALL_ and _GUESTS_ group.

**Comment by olee** [comment]
Spawn problem will be fixed as part of #1612
Please only continue discussion of the problem with explosions here.

**Comment by Mansarde** [comment]
At first I thought I was going insane because the explosion protection seemed to work with some zones but not others. After enabling debugging I did the following:

1. Go inside a malfunctioning protection zone.
2. **`/p test fe.protection.explosion`** -> Result: **`false`** (which was expected)
3. Fire up some TNT, which then exploded (not expected)
The debug log showed that FE checked for **`fe.protection.explosion`** within **`[_ROOT_]`** instead of the custom zone.

After a sudden idea and some experimentation I finally figured out what the problem seemed to be.
It wasn't about any permissions or any wrong setup. It was about **the distance of the explosion to the zone border**.
I.e. if the explosion's *to-be-radius* would reach outside of the zone border then it seem that the permission was checked against the (higher-prio) outside world.
So, even though the explosion starts within the zone, one has to make sure that the maximum radius of it is *also* contained within the zone.

**tl;dr**
If you're someone wondering why your zones seem to disregard the explosion protection:
The would-be explosion radius reaches outside the zone, add some extra "buffer" to the borders of it. ^_^

**Comment by olee** [comment]
Mansarde hey that is a nice found!
It is supposed to be the other way round - if the explosion reaches even a bit into a protected area, it is supposed to stop the explosion. I will verify and fix this!

**Comment by Mansarde** [comment]
@olee You're welcome, glad to help. Even if it was just by describing a bug.^^
Really like all the stuff you put in FE, especially the permission system.
Thx for the continued support! :+1:
